src/main/scala/it/pagopa/interop/partyprocess/service/impl/PartyProcessMailer.scala

```diff
@@ -7,15 +7,20 @@ import it.pagopa.interop.partyprocess.service.MailEngine
 
 import java.io.File
 import java.nio.file.Files
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
+import com.typesafe.scalalogging.Logger
+import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
 
 /** Decorates mail-manager mailer with party-process features
   */
 trait PartyProcessMailer extends MailEngine { interopMailer: DefaultInteropMailer =>
+  implicit val logger = Logger.takingImplicit[ContextFieldsToLog](this.getClass())
 
   def sendMail(
     mailTemplate: PersistedTemplate
-  )(addresses: Seq[String], file: File, bodyParameters: Map[String, String]): Future[Unit] = {
+  )(addresses: Seq[String], file: File, bodyParameters: Map[String, String])
+  (emailPurpose: String = "email-notification")
+  (implicit executor: ExecutionContext, contexts: Seq[(String, String)]): Future[Unit] = {
```

Comment:
Trait definition MailEngine.sendMail must be changed by adding `executor: ExecutionContext`.